.circleci/config.yml

```
@@ -114,6 +117,65 @@ jobs:
               docker push ${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}:${CIRCLE_TAG}
             fi
 
+  proxy-build:
+    executor: docker/machine
+    steps:
+      - checkout
+      - run:
+          name: Generate build version.json
+          command: >
+            printf
+            '{"commit":"%s","version":"%s","source":"https://github.com/%s/%s","build":"%s"}\n'
+            "$CIRCLE_SHA1" "$CIRCLE_TAG" "$CIRCLE_PROJECT_USERNAME"
+            "$CIRCLE_PROJECT_REPONAME" "$CIRCLE_BUILD_URL"
+            > version.json
+      - run:
+          name: Output version.json
+          command: cat version.json
+      - docker/build:
+          path: dockerfiles/proxy
+          image: ${CIRCLE_PROJECT_REPONAME}-auth0-proxy
+          tag: ${CIRCLE_SHA1:0:9}
+      - run:
+          name: docker save
+          command: |
+            mkdir -p /tmp/cache
+            docker save -o /tmp/cache/auth0-proxy.tar "auth0-proxy:build"
+      - persist_to_workspace:
+          root: /tmp/cache
+          paths:
+            - auth0-proxy.tar
+
+  proxy-publish:
+    docker:
+      - image: cimg/base:stable
+        auth:
+          username: $DOCKER_USER
+          password: $DOCKER_PASSWORD
+    steps:
+      - setup_remote_docker
+      - restore_cache:
+          key: v1-{{.Environment.CIRCLE_SHA1}}
+      - run:
+          name: Restore Docker image cache
+          command: docker load -i /tmp/cache/auth0-proxy.tar
+      - run:
+          name: Login to GAR
```

```suggestion
          name: Login to GCR
```
This is using Google Container Registry (GCR) rather than Google Artifact Registry (GAR)